cmip6_downscaling/methods/deepsd/tasks.py

```diff
@@ -0,0 +1,475 @@
+    output_node_name,
+    res_to_str,
+    stacked_model_path,
+    starting_resolutions,
+)
+
+scratch_dir = UPath(config.get("storage.scratch.uri"))
+intermediate_dir = UPath(config.get("storage.intermediate.uri")) / version
+results_dir = UPath(config.get("storage.results.uri")) / version
+use_cache = config.get('run_options.use_cache')
+
+xr.set_options(keep_attrs=True)
+
+
+@task(log_stdout=True)
+def shift(path: UPath, path_type: str, run_parameters: RunParameters) -> UPath:
+    """Interpolate obs data to grid specs in ``xe.util.grid_global``.
+
+    Parameters
+    ----------
+    path : UPath
+        Path to original dataset.
+    path_type : str
+        Specify whether gcm or obs to set output resolution
+    run_parameters : RunParameters
+        Parameters for run set-up and model specs
+
+    Returns
+    -------
+    UPath
+        Path to shifted dataset.
+    """
+    if path_type == 'obs':
+        output_degree = 0.25
+    elif path_type == 'gcm':
+        output_degree = starting_resolutions[run_parameters.model]
+    else:
+        raise ValueError('path_type must be gcm or obs')
+
+    ds_hash = str_to_hash(str(path) + str(output_degree))
+    target = intermediate_dir / 'shift' / ds_hash
+
+    if use_cache and zmetadata_exists(target):
+        print(f'found existing target: {target}')
+        shifted_ds = xr.open_zarr(target)
+        return target
+
+    orig_ds = xr.open_zarr(path)
+
+    # Note: Notebook included unit conversion at this step for precipitation, but this is done earlier in `load_cmip`
+    shifted_ds = bilinear_interpolate(ds=orig_ds, output_degree=output_degree)
+    shifted_ds.attrs.update({'title': 'shift'}, **get_cf_global_attrs(version=version))
+    print(f'writing shifted dataset to {target}')
+    shifted_ds.pipe(set_zarr_encoding).to_zarr(target, mode='w')
+    return target
+
+
+@task(log_stdout=True)
+def coarsen_obs(path: UPath, output_degree: float) -> UPath:
+    """Coarsen grid using conservative interpolation.
+
+    Parameters
+    ----------
+    path : UPath
+        Path to original (likely observational) dataset.
+    output_degree : float
+        Resolution to coarsen the dataset to.
+
+    Returns
+    -------
+    UPath
+        Path to coarsened dataset.
+    """
+    # Similar to coarsen_and_interpolate in GARD tasks (maybe could be combined?)
+    ds_hash = str_to_hash(str(path) + str(output_degree))
+    target = intermediate_dir / 'coarsen' / ds_hash
+
+    if use_cache and zmetadata_exists(target):
+        print(f'found existing target: {target}')
+        coarse_ds = xr.open_zarr(target)
+        return target
+
+    orig_ds = xr.open_zarr(path)
+
+    # Coarsen obs
+    coarse_ds = conservative_interpolate(ds=orig_ds, output_degree=output_degree)
+    coarse_ds.attrs.update({'title': 'coarsen_interpolate'}, **get_cf_global_attrs(version=version))
+    print(f'writing coarsened dataset to {target}')
+    coarse_ds.pipe(set_zarr_encoding).to_zarr(target, mode='w')
```

Comment:
Same suggestion as in https://github.com/carbonplan/cmip6-downscaling/pull/239/files#r906733871